### ISSUE TITLE
1.19.2 Geyser has extremely high latency

### ISSUE THREAD

**Issue by Aldebaran-0** [GeyserMC/Geyser#3233]
### Describe the bug

Ever since I updated my server to 1.19.2 and updated geyser, performance for MCBE has been awful. If you stand still for a bit the lag eases out but whenever you load new chunks, break or place blocks, encounter a mob or try to interact with items (eat/shoot bow) or villagers, the game has the same performance you'd see if your ping was awful. Rubberbanding everywhere and the game catches up after about 2 minutes of lagging and a still screen. This was not the case on 1.19.1, and by downgrading a test server to 1.19.1 performance was restored to the good performance i was used to. Once on 1.19.2 however, this begins to happen.

### To Reproduce

I see this hasn't been reported much before so I am not sure how well this can be reproduced on other servers but..
1. Try a 1.19.1 server with geyser
2. Try a 1.19.2 server with geyser
3. Compare

### Expected behaviour

Normal performance like 1.19-1.19.1 had.

### Screenshots / Videos

Performance on 1.19.1 (Same VPS, same server software) vs 1.19.2

1.19.1 - Stable ping, can hit mobs, can break blocks, purpur compass consistently works.
https://user-images.githubusercontent.com/109926207/184544877-13dec1a0-29e0-4ce1-a02c-9c3dacd51fef.mp4

(Terribly sorry for the poor recording, at least i hope it shows the issue)

1.19.2 - Blocks don't drop when broken, Can't hit mobs, can't send messages or commands in chat, compass stuck. This would all fix itself after maybe 2 minutes of standing still everything would speed up and catch up with the server but if i move or do anything after it catches up it immediately falls back again.
https://user-images.githubusercontent.com/109926207/184544912-2a60f09a-03cf-44ac-b2c5-cc3f2f89533e.mp4

Hopefully this helps with identifying anything
Plugins being removed did not affect the lag on the 1.19.2 server, neither did datapacks
Messed with the config a lot, no avail.

### Server Version and Plugins

On the 1.19.2 server
git-purpur-1759
Latest 1.19.2

Plugins : 
Geyser
Floodgate
FastAsyncWorldEdit + VoxelSniper
MyWorlds (MultiVerse replacement)
Protocollib
BKCommonLib
Libsdisguises
AdvancedVanish
Invsee++
InventoryRollback+
Coreprotect

### Geyser Dump

https://dump.geysermc.org/FsvZvukCm294alh4wMPD4dO9vXSCLsq0

### Geyser Version

2.0.7-SNAPSHOT (git-master-33af9e0) (Latest for 1.19.1-1.19.2)

### Minecraft: Bedrock Edition Device/Version

1.19.20 - PC Windows10 Edition

**Comment by Camotoy** [comment]
1.19.2 was a VERY small update - so small that Geyser works still on both 1.19.1 and 1.19.2. 
This is Purpur's last 1.19.1 build: https://api.purpurmc.org/v2/purpur/1.19.1/1751/download
And their first 1.19.2 build: https://api.purpurmc.org/v2/purpur/1.19.2/1752/download

Is this performance difference observable between these two builds? Does it happen on Paper? On Spigot?

**Comment by Aldebaran-0** [comment]
Yes, the videos i sent are the performance of geyser's last build before 1.19.2 on purpur 1751 1.19.1 vs geyser latest build on purpur 1752 1.19.2

**Comment by Camotoy** [comment]
(Also, if you test Paper/Spigot, please also test Pufferfish)

**Comment by Aldebaran-0** [comment]
On pufferfish, paper and spigot 1.19.1 geyser performance is good.
On pufferfish, paper and spigot 1.19.2 geyser performance is as bad as purpur 1.19.2, which is why my last resort has been sending an issue here.

**Comment by Aldebaran-0** [comment]
I also started on Petal, a fork of purpur which is said to improve performance or whatev, but same issue there too.

**Comment by Camotoy** [comment]
Do these performance issues happen if you use Geyser standalone? What about without using Floodgate?

**Comment by Aldebaran-0** [comment]
Issue persists on standalone 1.19.2 geyser

**Comment by alebastroenm** [comment]
Good afternoon, a user on Discord (maybe a mod or contributor) asked me to attach an argument here so I will.

Yesterday I updated my Java server to 1.19.20, also updated Geyser with support for this version. All this under Paper software.

I had the same problem as the author of this thread. Intense lag, using only two plugins: Bass, and Harbor.

Since half of my server are Bedrock users, I decided to check for several hours what was the cause of the lag, and I didn't find a solution, even removing the plugins and resetting the server completely. Then opt to downgrade the server version (from 1.19.20 to 1.19.10). This pretty much fixed the server lag. This using the latest Geyser build (1175). Geyser (when i tried to connect to server) recommend using ViaVersion, so I quickly installed the latest version and it worked correctly, everything, without lag.

Still, one of my Java users asked if I could downgrade to 1.19 (for cosmetic reasons), so I also downgraded my world (from 1.19.10 to 1.19.0), keeping the ViaVersions and Geyser on its latest version.

I must admit this fixed a LOT MORE lag for some reason (always 20 TPS, much lower ping than I got with 1.19.10, etc).

In a nutshell: if you use the latest build of Geyser (1175) on a JE 1.19.0 server with ViaVersions, the difference in lag is huge (maybe it sounds exaggerated, but it changes a lot).

**Comment by alebastroenm** [comment]
In addition to that, it is worth noting that you do not use Floodgate for the 1.19.0 server. Maybe later try, although I doubt anything will change.

I tried Floodgate on 1.19.20, and with or without this, the server lags.

I tried Floodgate on 1.19.10 and there were no problems.

**Comment by onebeastchris** [comment]
Does this issue still occur with the latest version of Geyser? Specifically, after the recent internal changes with the protocol-3.0 merge - simply update to the latest version to test.

**Comment by onebeastchris** [comment]
Closing due to no response.

**Comment by makcuber** [comment]
Not sure if this issue has been addressed elsewhere already, but Im getting the same reported lag issues with Geyser running on Waterfall in front of a Paper 1.20.1 server with floodgate running on both Waterfall and Paper.

The moment I connect a Bedrock player to the server there is a massive lag spike, TPS tanks and the server cant keep up even on solid hardware. The lag persists even after the bedrock player disconnects, and is only fixed by restarting the server. If I remove geyser+floodgate and just connect a Java player there is no lag whatsoever.

I noted from another thread about this issue that others noticed the lag is especially bad whenever the bedrock player interacts or even moves along a block, and in my testing I found the same result. Theres a massive lag spike when the bedrock player connects, but if they dont move at all the lag stabilizes a bit, but then as soon as they move the lag is even worse that the initial connection. Others suggested that the issue is somewhere in the packet translation between Java and Bedrock, specifically the differences in the coordinate system, but im not setup to verify that currently.

Hopefully this issue is resolved or being looked at, because as for now I simply cannot use geyser on my server so no bedrock players can connect until I find a fix for this.
